src/main/java/cms/gov/madie/measure/clients/UserServiceClient.java

```diff
@@ -0,0 +1,89 @@
+package cms.gov.madie.measure.clients;
+
+import gov.cms.madie.models.dto.DetailsRequestDto;
+import gov.cms.madie.models.dto.UserDetailsDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class UserServiceClient {
+
+  private final RestTemplate userServiceRestTemplate;
+
+  @Value("${madie.user-service.base-url}")
+  private String userServiceBaseUrl;
+
+  /**
+   * Fetches user details in bulk from the user service.
+   *
+   * @param harpIds List of HARP IDs to fetch details for
+   * @return Map of HARP ID to UserDetailsDto, empty map if service call fails
+   */
+  public Map<String, UserDetailsDto> getBulkUserDetails(List<String> harpIds) {
+    if (CollectionUtils.isEmpty(harpIds)) {
+      return Collections.emptyMap();
+    }
+
+    try {
+      String url = userServiceBaseUrl + "/users/details";
+
+      HttpHeaders headers = new HttpHeaders();
+      headers.setContentType(MediaType.APPLICATION_JSON);
+
+      DetailsRequestDto requestBody = DetailsRequestDto.builder().harpIds(harpIds).build();
+
+      HttpEntity<DetailsRequestDto> request = new HttpEntity<>(requestBody, headers);
+
+      log.debug("Requesting user details for HARP IDs: {}", harpIds);
+
+      ResponseEntity<Map<String, UserDetailsDto>> responseEntity =
+          userServiceRestTemplate.exchange(
+              url,
+              HttpMethod.POST,
+              request,
+              new ParameterizedTypeReference<Map<String, UserDetailsDto>>() {});
+
+      Map<String, UserDetailsDto> response = responseEntity.getBody();
+
+      log.debug("Received response from user service: {}", response);
+
+      if (response != null) {
```

Comment:
Trying to avoid running the loop if debug is disabled. This was the simplest patternI could think of.
```suggestion
      if (response != null && log.isDebugEnabled()) {
```